File: src/workflows/workflowSemanticAlignmentEngine.ts
/**
 * Converted from Python: core/workflows/workflow_semantic_alignment_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function alignWorkflowSemantics(plan: any, semantic_runtime: any = null, causality: any = null): any {
  semantic_runtime = py.or2(semantic_runtime, () => ({}));
  var inner: any = py.get(semantic_runtime, "semantic", semantic_runtime);
  return {"ontology": py.get(inner, "ontology", {}), "domain": py.get(inner, "domain", {}), "causality_chains": (py.truthy(causality) ? py.get(py.get(causality, "causality", causality), "propagation", {}) : {}), "workflow_objective": py.get(plan, "objective", ""), "extraction_semantics": py.get(inner, "semantic_graph", {}), "aligned": true, "bounded": true};
}

File: lib/src/workflows/workflow_semantic_alignment_engine.dart
// Port of core/workflows/workflow_semantic_alignment_engine.py

Map<String, dynamic> alignWorkflowSemantics(
  Map<String, dynamic> plan, {
  Map<String, dynamic>? semanticRuntime,
  Map<String, dynamic>? causality,
}) {
  final Map<String, dynamic> semantic = semanticRuntime ?? <String, dynamic>{};
  final dynamic innerRaw = semantic['semantic'];
  final Map<String, dynamic> inner =
      innerRaw is Map ? Map<String, dynamic>.from(innerRaw) : semantic;

  dynamic causalityChains = <String, dynamic>{};
  if (causality != null && causality.isNotEmpty) {
    final dynamic cInnerRaw = causality['causality'];
    final Map<String, dynamic> cInner =
        cInnerRaw is Map ? Map<String, dynamic>.from(cInnerRaw) : causality;
    causalityChains = cInner['propagation'] ?? <String, dynamic>{};
  }

  return <String, dynamic>{
    'ontology': inner['ontology'] ?? <String, dynamic>{},
    'domain': inner['domain'] ?? <String, dynamic>{},
    'causality_chains': causalityChains,
    'workflow_objective': plan['objective'] ?? '',
    'extraction_semantics': inner['semantic_graph'] ?? <String, dynamic>{},
    'aligned': true,
    'bounded': true,
  };
}
